File: apps/web/src/types/skill.ts
/**
 * File purpose: Defines shared TypeScript contracts for skill.
 * Main declarations: this file contains declarative configuration or re-exports and has no
 * callable declarations.
 */

export type SkillKey =
  | "data_analysis"
  | "deep_research"
  | "html_generation"
  | "ppt_generation"
  | "u1_image";

export interface Skill {
  key: SkillKey;
  name: string;
  description: string;
  version: string;
  enabled: boolean;
  isDefault?: boolean;
  lastUpdatedAt?: string;
}

File: services/api/app/schemas/skill.py
# File purpose: Defines Pydantic API contracts for skill.
# Main declarations: Skill defines skill state or behavior.

from app.schemas.base import ApiModel
from app.schemas.session import SkillKey


class Skill(ApiModel):
    key: SkillKey
    name: str
    description: str
    version: str
    enabled: bool
    is_default: bool | None = None
    last_updated_at: str | None = None
